gtfs/__main__.py

```diff
@@ -1,25 +1,42 @@
 #!/usr/bin/env python
 """Command line interface for fetching GTFS."""
-import argparse
-import getpass
+import typer
+from chalky import chain
 import logging
-import sys
-
 from prettytable import PrettyTable
-
-from gtfs.utils.FeedSource import FeedSource
+from halo import Halo
+import importlib
+from utils.FeedSource import FeedSource
 import feed_sources
-# import all the available feed sources
-# pylint: disable=I0011,wildcard-import
-from feed_sources import *
 
 logging.basicConfig()
 LOG = logging.getLogger()
 LOG.setLevel(logging.INFO)
+app = typer.Typer()
+sources = list(feed_sources.__all__)
+            
+@app.command()
+def list_feeds(xmin:float, ymin:float, xmax:float, ymax:float):
+    """Filter feeds spatially based on bounding box. Pass negative args as ' -45.6774' for example."""
+    spinner = Halo(text='Filtering...', text_color="cyan", spinner='dots')
+    spinner.start()
+    ptable = PrettyTable(['Transit URL', 'Bounding Box'])
+    
+    for src in sources:
+        module = importlib.import_module('feed_sources.' + src)
+        klass = getattr(module, src)
+        inst = klass()
+        bbox = inst.bbox
+        if (bbox[0] >= xmin and bbox[0] < xmax) and (bbox[1] >= ymin and bbox[1] < ymax) and (bbox[2] <= xmax and bbox[2] > xmin) and (bbox[3] <= ymax and bbox[3] > ymin):
+            row=[ chain.bright_yellow | inst.url , bbox ]
+            ptable.add_row(row)
+    print(f"\n Feeds based on bbox input {chain.bright_yellow | [xmin, ymin, xmax, ymax]} are as follows:")     
+    print("\n" + ptable.get_string())
+    spinner.succeed(chain.bright_green.bold | "All done!")
```

Comment:
I can appreciate the UX experience these things give. However, the result of this command should also be machine-readable, i.e. a user should be able to use the output of this function directly to actually download the feeds eventually. Maybe some arg like `--pretty` could be introduced later to give control over that.

Reply:
So what kind of an output do u expect? Like I am not talking about the spinners etc but like with the table's output?

Reply:
I think it's nice to have a human-readable table (what you did already). But it also needs a machine-readable version, which just prints out the class names which we'll expect for a future `gtfs_fetcher fetch` command. I'd keep this in an issue for later, once we get to the fetch command.

Reply:
Ahh okay got it 👍